client/state/data-layer/wpcom/comments/index.js

```diff
@@ -58,6 +117,16 @@ export const addComments = ( { dispatch }, { siteId, postId }, next, { comments,
 	}
 };
 
+export const writePostCommentSuccess = ( { dispatch, getState }, { siteId, postId, parentCommentId, placeholderId }, next, comment ) => {
+	// remove placeholder from state
+	dispatch( { type: COMMENTS_REMOVE, siteId, postId, commentId: placeholderId } );
+	// add new comment to state with updated values from server
+	dispatch( { type: COMMENTS_RECEIVE, siteId, postId, comments: [ comment ], skipSort: !! parentCommentId } );
+	// increment comments count
+	const commentCount = getPostTotalCommentsCount( getState(), siteId, postId );
+	dispatch( { type: COMMENTS_COUNT_RECEIVE, siteId, postId, totalCommentsCount: commentCount + 1 } );
```

Comment:
I think this is OK. I just wanted to point out the old write action triggers another network request to check for comment counts at this point if folks see any differences in behavior.